addons/mail/models/res_users_settings.py

```diff
@@ -25,6 +61,16 @@ def _format_settings(self, fields_to_format):
             res['volumes'] = [('ADD', volume_settings)]
         return res
 
+
+    def write(self, vals):
+        write_res = super(ResUsersSettings, self).write(vals)
+        if 'notification_type' in vals:
+            self.env['bus.bus']._sendmany([
+                (setting.user_id.partner_id, 'mail.record/insert', {'Persona': setting.user_id.partner_id.mail_partner_format().get(setting.user_id.partner_id)})
+                for setting in self
+            ])
```

Comment:
The current one may send notification to the same partner multiple times? Not sure if you want it.
```suggestion
            self.env['bus.bus']._sendmany([
                (partner, 'mail.record/insert', {'Persona': partner.mail_partner_format()[partner]})
                for partner in self.user_id.partner_id
            ])
```
I guss `mail_partner_format` can prefetch everything as expected, otherwise it would be better to do it in batch